src/main/java/use_case/search/SearchInteractor.java

```diff
@@ -0,0 +1,75 @@
+package use_case.search;
+
+import data_access.movies.MovieDBDataAccessException;
+import entity.Movie;
+import use_case.generate_recommendations.MovieDBDataAccessInterface;
+
+/**
+ * Interactor for the search by criteria use case.
+ */
+public class SearchInteractor implements SearchInputBoundary {
+    private final MovieDBDataAccessInterface movieDBDataAccess;
+    private final SearchOutputBoundary outputBoundary;
+
+    public SearchInteractor(MovieDBDataAccessInterface movieDBDataAccess, SearchOutputBoundary outputBoundary) {
+        this.movieDBDataAccess = movieDBDataAccess;
+        this.outputBoundary = outputBoundary;
+    }
+
+    @Override
+    public void execute(SearchByCriteriaInputData inputData) {
+        Movie matchingMovie = null;
+        String movieDetails;
+
+        // Check if the list of keywords is empty or if the first keyword is empty or contains only whitespace
+        if (inputData.getKeywords().isEmpty() || inputData.getKeywords().get(0).trim().isEmpty()) {
+            movieDetails = "Error: Please provide a valid movie title.";
+            outputBoundary.prepareFailView(new SearchByCriteriaOutputData(movieDetails));
+        }
+        else {
+            try {
+                // Validate the category (Movie/TV show)
+                if (validateCategory(inputData.getCategory())) {
+                    final String searchKeyword = inputData.getKeywords().get(0).trim();
+
+                    // Call DAO to search for the movie if a valid keyword is provided
+                    if (!searchKeyword.isEmpty()) {
+                        matchingMovie = movieDBDataAccess.getMovie(searchKeyword);
+                    }
+                }
+
+                // Convert result to output data
+                if (matchingMovie != null) {
+                    movieDetails = matchingMovie.toString();
+                }
+                else {
+                    movieDetails = "No results found for the keyword: " + inputData.getKeywords().get(0);
+                }
```

Comment:
ah, instead of wrapping validateCategory inside a helper method, it should be the body of the if block that gets extracted into a helper method
so it'd be like if the category is movie, do the searchMovie (helper) method, which contains the highlighted code right now.
else if the category is tv, throw a UnsupportedOperationException for now, or just go to the failview saying tv shows aren't currently supported, and do the same in the final else statement. 
The code as is right now would just return "No results found for the keyword" when you try to click search by keyword on the tv category when that's not what should happen.

Reply:
I extracted it into a helper method but for some reason it's still going through with the movie search when you switch the category to TV show

Reply:
Ah, since in SearchController, "movie" was hardcoded into the input data.